src/hooks/useSingleCampaignHandler.ts

```diff
@@ -73,6 +78,43 @@ export const useSingleCampaignHandler = () => {
     }
   }, [DAOSEEDER_FACTORY_ADDRESS, id, provider]);
 
+  useEffect(() => {
+    const fetchStageData = async () => {
+      if (campaign && campaign.stageCount > 0 && DAOSEEDER_FACTORY_ADDRESS) {
+        const contract = await getSmartContractWithProvider(
+          DAOSEEDER_FACTORY_ADDRESS,
+          provider,
+          JSON.stringify(DaoSeederFactory.abi)
+        );
+        const obj: ICampaignStage[] = [];
+        for (let i = 0; i < campaign.stageCount; i++) {
+          const stageKey = await contract.getKey(campaign.tokenAddress, i);
+          const stage = await contract.getStage(stageKey);
+          if (constants.AddressZero !== stage) {
+            const stageContract = await getSmartContractWithProvider(
+              stage,
+              provider,
+              JSON.stringify(StageContract.abi)
+            );
+            const stageIpfsKey = await stageContract.ipfsKey();
+            const stageIpfsData = await getStageData(stageIpfsKey);
+            obj.push({ name: stageIpfsData.name, address: stage });
```

Comment:
if this is long, then just show something like pagination buttons like this:
https://miro.medium.com/v2/resize:fit:720/format:webp/1*OjTTU_GBL1bLgnVTAmA-KA.png
so no need to fetch the name.

Reply:
Done